File: ios/Classes/Tasks/LastKnownLocationTask.m
//
//  LastKnownLocationTask.m
//  geolocator
//
//  Created by Maurits van Beusekom on 20/05/2019.
//

#import "LastKnownLocationTask.h"


@implementation LastKnownLocationTask

- (instancetype)initWithContext:(TaskContext *)context completionHandler:(CompletionHandler)completionHandler {
    self = [super initWithContext:context completionHandler:completionHandler];
    if (self) {
        //
    }
    
    return self;
}

- (void)startTask {
    CLLocationManager *locationManager = [CLLocationManager new];
    CLLocation *location = [locationManager location];
    
    if (location == nil) {
        [[self context] resultHandler](nil);
        return;
    }
    
    [[self context] resultHandler]([location toDictionary]);
    [self stopTask];
}

@end

File: ios/Classes/Tasks/LastKnownLocationTask.swift
//
//  CalculateDistance.swift
//  geolocator
//
//  Created by Maurits van Beusekom on 18/07/2018.
//

import Foundation
import CoreLocation

class LastKnownLocationTask : Task, TaskProtocol {
    required init(context: TaskContext, completionHandler: CompletionHandler?) {
        super.init(context: context,
                   completionHandler: completionHandler)
    }
    
    func startTask() {
        let locationManager = CLLocationManager.init()
        let location = locationManager.location;
        
        guard let lastKnownPosition = location?.toDictionary() else {
            context.resultHandler(nil)
            return
        }

        context.resultHandler(lastKnownPosition)
        
        stopTask()
    }
}
